src/language/sharedLanguage.ts

```diff
@@ -23,33 +23,48 @@ DOMPurify.addHook('afterSanitizeAttributes', (node) => {
   }
 });
 
-export function getLanguageFromKey(key: string | undefined, language: ILanguage) {
+/**
+ * @deprecated Use lang() from useLanguage.ts instead
+ * @see useLanguage
+ */
+export function getLanguageFromKey(key: string | undefined, language: ILanguage | null) {
   if (!key) {
     return key;
   }
-  const name = getNestedObject(language, key.split('.'));
+  const name = language ? getNestedObject(language, key.split('.')) : undefined;
   if (!name) {
     return key;
   }
   return name;
 }
 
-export function getNestedObject(nestedObj: any, pathArr: string[]) {
+function getNestedObject(nestedObj: any, pathArr: string[]) {
```

Comment:
What about using unknown instead of any?

Alternative could be using generics, but I think that might be "overkill" 😄 
```suggestion
function getNestedObject<T>(nestedObj: T, pathArr: string[]) {
```

Reply:
Oh boy, that really opened the hornets nest! I just un-exported the function because it wasn't used anywhere else, but adding the proper type (`ILanguage`) as the input caused the return type to possibly be an object. Found a case in the List component that relied on it being an object, but I got that cleaned up and introduced even more proper types everywhere. Now there's auto-completion for language keys pretty much all over the place! 🥳